components/brave_wallet_ui/page/screens/swap/utils.ts

```diff
@@ -0,0 +1,193 @@
+      minimumToAmount: undefined,
+      fromToken,
+      toToken,
+      rate: new Amount(quote.buyAmount)
+        .divideByDecimals(toToken.decimals)
+        .div(new Amount(quote.sellAmount).divideByDecimals(fromToken.decimals)),
+      impact: new Amount(quote.estimatedPriceImpact),
+      sources: quote.sources
+        .map((source) => ({
+          name: source.name,
+          proportion: new Amount(source.proportion)
+        }))
+        .filter((source) => source.proportion.gt(0)),
+      routing: 'split', // 0x supports split routing only
+      networkFee,
+      networkFeeFiat: networkFee.isUndefined()
+        ? ''
+        : networkFee
+            .times(
+              getTokenPriceAmountFromRegistry(
+                spotPrices,
+                makeNetworkAsset(fromNetwork)
+              )
+            )
+            .formatAsFiat(defaultFiatCurrency)
+    }
+  ]
+}
+
+function getJupiterNetworkFee({
+  quote,
+  fromNetwork
+}: {
+  quote: BraveWallet.JupiterQuote
+  fromNetwork: BraveWallet.NetworkInfo
+}): Amount {
+  if (!fromNetwork) {
+    return Amount.empty()
+  }
+
+  return new Amount('0.000005')
+}
+
+export function getJupiterFromAmount({
+  quote,
+  fromToken
+}: {
+  quote: BraveWallet.JupiterQuote
+  fromToken: BraveWallet.BlockchainToken
+}): Amount {
+  return new Amount(quote.inAmount).divideByDecimals(fromToken.decimals)
+}
+
+export function getJupiterToAmount({
+  quote,
+  toToken
+}: {
+  quote: BraveWallet.JupiterQuote
+  toToken: BraveWallet.BlockchainToken
+}): Amount {
+  return new Amount(quote.outAmount).divideByDecimals(toToken.decimals)
+}
+
+export function getJupiterQuoteOptions({
+  quote,
+  fromNetwork,
+  fromToken,
+  toToken,
+  spotPrices,
+  defaultFiatCurrency
+}: {
+  quote: BraveWallet.JupiterQuote
+  fromNetwork: BraveWallet.NetworkInfo
+  fromToken: BraveWallet.BlockchainToken
+  toToken: BraveWallet.BlockchainToken
+  spotPrices: SpotPriceRegistry
+  defaultFiatCurrency: string
+}): QuoteOption[] {
+  const networkFee = getJupiterNetworkFee({ quote, fromNetwork })
+
+  return [
+    {
+      label: '',
+      fromAmount: new Amount(quote.inAmount).divideByDecimals(
+        fromToken.decimals
+      ),
+      toAmount: getJupiterToAmount({ quote, toToken }),
+      // TODO: minimumToAmount is applicable only for ExactIn swapMode.
+      // Create a maximumFromAmount field for ExactOut swapMode if needed.
+      minimumToAmount: new Amount(quote.otherAmountThreshold).divideByDecimals(
+        toToken.decimals
+      ),
+      fromToken,
+      toToken,
+      rate: new Amount(quote.outAmount)
+        .divideByDecimals(toToken.decimals)
+        .div(new Amount(quote.inAmount).divideByDecimals(fromToken.decimals)),
+      impact: new Amount(quote.priceImpactPct),
+      sources: [
+        ...new Set(quote.routePlan.map((step) => step.swapInfo.label))
+      ].map((name) => ({
+        name,
+        proportion: new Amount(1)
```

Comment:
can we comment as to why the proportion is always `1`?

Reply:
https://github.com/brave/brave-core/pull/22285/commits/075add5075d53ced95d2c763d814318197415abc